shell/platform/android/external_view_embedder/external_view_embedder.cc

```diff
@@ -142,8 +144,14 @@ bool AndroidExternalViewEmbedder::SubmitFrame(
     background_canvas->drawPicture(pictures.at(view_id));
   }
   // Submit the background canvas frame before switching the GL context to
-  // the surfaces above.
-  frame->Submit();
+  // the overlay surfaces.
+  //
+  // Skip a frame if the embedding is switching surfaces.
+  auto should_submit_current_frame =
+      previous_frame_view_count_ > 0 || current_frame_view_count == 0;
+  if (should_submit_current_frame) {
```

Comment:
this is `false`, when the Java side is switching from `FlutterSurfaceView` to `FlutterImageView`.